modules/API/pages/upsert-rest.adoc

```diff
@@ -314,26 +315,175 @@ curl -X POST "localhost:14240/restpp/graph/Person_Movie" -d '
 '
 ----
 
+== Upserting regular edges 
+
+Regular edges do not have discriminators and must be uniquely defined by their source and target vertex IDs.
+To upsert a regular edge, use the following JSON format:
+
+[source,json]
+----
+{
+  "edges": {
+    "<source_vertex_type>": {
+      "<source_vertex_id>": {
+        "<edge_type>": {
+          "<target_vertex_type>": {
+            "<target_vertex_id>": {
+              "<attribute>": {
+                "value": <value>
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+}
+----
+
+=== Examples
+
+Upserting a `Liked` edge from a `User` vertex (`id1`) to another `User` vertex (`id6`):
+
+[source,json]
+----
+{
+  "edges": {
+    "User": {
+      "id1": {
+        "Liked": {
+          "User": {
+            "id6": {
+              "weight": {
+                "value": 5.0
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+}
+----
+
 == Upserting edges with discriminators
 
-Some edge types are defined with xref:gsql-ref:ddl-and-loading:defining-a-graph-schema.adoc#_discriminator[discriminators], which allow multiple instances of the same edge type between two vertices.
+Some edge types are defined with xref:gsql-ref:ddl-and-loading:defining-a-graph-schema.adoc#_discriminator[discriminators], which allow multiple instances of the same edge type between two vertices. 
+
+=== Rules for upserting edges with discriminators
 
-To upsert an edge that was defined with a discriminator, insert them as a regular edge.
-However, the following rules apply:
+1. *Discriminator attributes are required:*
+* You must include all attributes defined in the discriminator when inserting an edge.
 
-* You cannot leave off discriminator attributes when inserting an edge whose type was defined with discriminator attributes.
-* If you are updating an existing edge, you cannot update the attributes that are defined as part of the edge type discriminator.
+2. *Discriminator attributes cannot be updated:*
+* Discriminator attributes are immutable and cannot be changed once the edge is created.
 
-For example, if you have the following edge type definition:
+3. *Support for multi-edges with JSON arrays:*
+* When upserting multiple edges with discriminators between the same source and target vertices, you can use a JSON array format..
+
+=== Example: Edge type definition
 
 [source.wrap,console]
 ----
-CREATE DIRECTED EDGE Study_At(From Person, To University, DISCRIMINATOR(class_year INT, class_month INT), major STRING))
+CREATE DIRECTED EDGE Liked (  
+  FROM User,  
+  TO User,  
+  DISCRIMINATOR(actionId STRING),  
+  weight FLOAT  
+);  
+----
+
+==== Old format (Single edge):
```

Comment:
We should avoid saying "new" and "old" unless we have clearly established in this context what historical event we are talking about.  You an explanation for why there are two formats. Also, I believe the developer added a comment recently that explained even in the "old" way, each edge of a multi-edge count in inserted one at a time, using standard JSON.